### PR TITLE
Fix assert for external interpreter

### PR DESCRIPTION
**Pull request by aaronj0**

### CODE REVIEW

File: unittests/CppInterOp/InterpreterTest.cpp [hunk 1]
```diff
@@ -200,7 +200,7 @@ TEST(InterpreterTest, ExternalInterpreterTest) {
 #endif
 
   EXPECT_NE(ExtInterp, nullptr);
-  Cpp::UseExternalInterpreter(ExtInterp);
+  ASSERT_DEATH(Cpp::UseExternalInterpreter(ExtInterp), "Existing Interpreter has been set");
```
Comment (vgvassilev) [inline unittests/CppInterOp/InterpreterTest.cpp:203]:
```suggestion
#if defined(NDEBUG) || !GTEST_HAS_DEATH_TEST
  ASSERT_DEATH(Cpp::UseExternalInterpreter(ExtInterp), "Existing Interpreter has been set");
#endif
```

File: unittests/CppInterOp/InterpreterTest.cpp [hunk 2]
```diff
@@ -200,7 +203,10 @@ TEST(InterpreterTest, ExternalInterpreterTest) {
 #endif
 
   EXPECT_NE(ExtInterp, nullptr);
-  Cpp::UseExternalInterpreter(ExtInterp);
+
+#if defined(NDEBUG) || !GTEST_HAS_DEATH_TEST
```
Comment (vgvassilev) [inline unittests/CppInterOp/InterpreterTest.cpp:207]: This needs to become `#if !defined(NDEBUG) || !GTEST_HAS_DEATH_TEST`

File: unittests/CppInterOp/InterpreterTest.cpp [hunk 3]
```diff
@@ -200,7 +203,13 @@ TEST(InterpreterTest, ExternalInterpreterTest) {
 #endif
 
   EXPECT_NE(ExtInterp, nullptr);
-  Cpp::UseExternalInterpreter(ExtInterp);
+
+#if !defined(NDEBUG) || !GTEST_HAS_DEATH_TEST
+  // death tests don't work right on windows
+  #ifndef _WIN32
```
Comment (vgvassilev) [inline unittests/CppInterOp/InterpreterTest.cpp:209]: I thought this was covered by `GTEST_HAS_DEATH_TEST`. Did anything in particular fail?

Comment (aaronj0) [reply unittests/CppInterOp/InterpreterTest.cpp:209]: Yes, it doesn't hit the assertion:
`  1: Actual msg:
  1: [  DEATH   ] Unable to find target for this triple (no targets are registered)
  `

Comment (aaronj0) [reply unittests/CppInterOp/InterpreterTest.cpp:209]: Shouldn't we do `#if !defined(NDEBUG) && !GTEST_HAS_DEATH_TEST`?

Comment (vgvassilev) [reply unittests/CppInterOp/InterpreterTest.cpp:209]: We do that, we do not need the `_WIN32` thing...

> Yes, it doesn't hit the assertion:
> 1: Actual msg: 1: [  DEATH   ] Unable to find target for this triple (no targets are registered)
 
 Seems a different problem...

File: unittests/CppInterOp/InterpreterTest.cpp [hunk 4]
```diff
@@ -200,7 +203,10 @@ TEST(InterpreterTest, ExternalInterpreterTest) {
 #endif
 
   EXPECT_NE(ExtInterp, nullptr);
-  Cpp::UseExternalInterpreter(ExtInterp);
+
+#if !defined(NDEBUG) && GTEST_HAS_DEATH_TEST
+    EXPECT_DEATH(Cpp::UseExternalInterpreter(ExtInterp), "sInterpreter already in use!");
```
Comment (vgvassilev) [inline unittests/CppInterOp/InterpreterTest.cpp:208]:
```suggestion
#ifndef _WIN32 // Windows seems to fail to die...
    EXPECT_DEATH(Cpp::UseExternalInterpreter(ExtInterp), "sInterpreter already in use!");
#endif // _WIN32
```

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Pull request merged**